### PR TITLE
Changes to adopt Consul Dataplane architecture for Consul ECS

### PR DESCRIPTION
**Pull request by Ganeshrockz**
## Changes proposed in this PR:
- With https://github.com/hashicorp/consul-ecs/pull/161 Consul-ecs migrate off of client agents and started using the Consul Dataplane architecture. This PR contains the changes needed for terraform submodules to support the same.
- The CHANGELOG should give an idea on the changes made to individual modules within this repo.

## How I've tested this PR:

CI, Manual deployment

## How I expect reviewers to test this PR:

Most of these files were already reviewed as part of previous PRs. I would like some 👀 on the CHANGELOG because that is the only major addition to the feature branch after all the other changes.

## Checklist:
- [X] Tests added
- [X] CHANGELOG entry added 

    [HashiCorp engineers only. Community PRs should not add a changelog entry.]::
    [Changelog entries should use present tense, e.g. "Add support for..."]::

### CODE REVIEW

File: modules/mesh-task/variables.tf [hunk 1]
```diff
@@ -136,22 +142,16 @@ variable "outbound_only" {
...
-  description = "Consul Docker image."
-  type        = string
-  default     = "public.ecr.aws/hashicorp/consul:1.15.4"
-}
-
 variable "consul_ecs_image" {
   description = "consul-ecs Docker image."
   type        = string
-  default     = "public.ecr.aws/hashicorp/consul-ecs:0.6.0"
+  default     = "hashicorpdev/consul-ecs:latest"
```
Comment (Ganeshrockz) [inline modules/mesh-task/variables.tf:148]: We'd ideally want to use the `hashicorppreview` version but we are unsure about the new consul ECS version yet. Until then we'll use the dev image

File: README.md [hunk 2]
```diff
@@ -5,30 +5,30 @@ AWS ECS (Elastic Container Service).
 
 ## Documentation
 
-See https://www.consul.io/docs/ecs for full documentation.
+See https://developer.hashicorp.com/consul/docs/ecs for full documentation.
 
 ## Architecture
 
-![Architecture](https://github.com/hashicorp/terraform-aws-consul-ecs/blob/main/_docs/architecture.png?raw=true)
+![Architecture](./_docs/architecture.png?raw=true)
```
Comment (cthain) [inline README.md:12]: This needs to be reverted to the full URL on main or the README won't display correctly on the Terraform rgy.

File: README.md [hunk 3]
```diff
@@ -5,30 +5,30 @@ AWS ECS (Elastic Container Service).
...
+  * After this, the container enters into its reconciliation loop where it periodically syncs the health of ECS containers into Consul.
+  * Upon receiving SIGTERM, it marks the corresponding service instance in Consul as unhealthy and waits for the dataplane container to shutdown.
+  * Finally, it deregisters the service and proxy entities from Consul's catalog and performs a Consul logout.
+* `consul-dataplane` – Runs for the full lifecycle of the task. This container runs
+  the [Consul dataplane](https://github.com/hashicorp/consul-dataplane) that configures and starts the Envoy proxy, which controls all the service mesh traffic. All requests to and from the application run through
+  the proxy.
 
-The `acl-controller` module runs a controller that automatically provisions ACL tokens
-for tasks on the mesh.
+The `ecs-controller` module runs a controller that automatically provisions ACL tokens
```
Comment (cthain) [inline README.md:30]: nit: the module is just called `controller`.

```suggestion
The `controller` module runs a controller that automatically provisions ACL tokens
```

File: README.md [hunk 4]
```diff
@@ -47,8 +47,8 @@ See https://www.consul.io/docs/ecs.
 * [dev-server](https://github.com/hashicorp/terraform-aws-consul-ecs/blob/main/modules/dev-server) [**For Development/Testing Only**]: This module deploys a Consul server onto your ECS Cluster
   for development/testing purposes. The server does not have persistent storage and so is not suitable for production deployments.
 
-* [acl-controller](https://github.com/hashicorp/terraform-aws-consul-ecs/blob/main/modules/acl-controller): This modules deploys a controller that automatically provisions ACL tokens
-  for services on the Consul service mesh.
+* [ecs-controller](https://github.com/hashicorp/terraform-aws-consul-ecs/blob/main/modules/ecs-controller): This modules deploys a controller that automatically provisions ACL tokens
```
Comment (cthain) [inline README.md:50]: updated the path to the controller module.

```suggestion
* [controller](https://github.com/hashicorp/terraform-aws-consul-ecs/blob/main/modules/controller): This modules deploys a controller that automatically provisions ACL tokens
```

**Review by cthain — APPROVED** [review]
Reviewed the CHANGELOG and README :+1:

All other files were reviewed in the PRs mentioned in the changelog.

**Pull request merged**